### PR TITLE
Migrated the existing Spring application from the previous module to …

### PR DESCRIPTION
**Pull request by Fat-Frumos**
…Spring Boot:

  - Updated dependencies in configuration files (pom.xml)
  - Removed AppConfig, RestConfig, WebAppInitializer
  - PersistenceConfig added
  - RestApiApplication starter was created.
  - java version updated to 1.8

### CODE REVIEW

**Comment by sslichnyi** [comment]
It will be good to create a new module 'starter' and put there main class with all properties (application.properties). And config package

**Comment by sslichnyi** [comment]
please, move db sql files to model module

**Comment by sslichnyi** [comment]
Please, remove spring-boot-starter-data-jpa dependency and use only hibernate

File: Module#03/api/src/main/java/com/epam/esm/RestApiApplication.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import com.epam.esm.config.PersistenceConfig;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.autoconfigure.domain.EntityScan;
+import org.springframework.context.annotation.Import;
+
+@SpringBootApplication
+@EntityScan(basePackages = "com.epam.esm.entity")
+@Import(PersistenceConfig.class)
+public class RestApiApplication {
```
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/RestApiApplication.java]: RestApiApplication Should be moved

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/RestApiApplication.java (hunk 1 above)
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/RestApiApplication.java]: Should be moved to a new module

**Comment by sslichnyi** [comment]
please, add IntelliJ Configuration files "Store as project file"

**Comment by sslichnyi** [comment]
please, edit application files in discussed way.

application.properties:
spring.datasource.url=jdbc:h2:mem:db
spring.datasource.username=user
spring.datasource.password=pass
spring.datasource.driverClassName=org.h2.Driver
spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.H2Dialect
spring.jpa.hibernate.ddl-auto=create
spring.jpa.show-sql=false

spring.mvc.static-path-pattern=/resources/**
spring.mvc.throw-exception-if-no-handler-found=true

spring.main.allow-circular-references=true
spring.main.banner-mode=off

server.tomcat.accesslog.enabled=true

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/config/PersistenceConfig.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        DriverManagerDataSource dataSource = new DriverManagerDataSource();
+        dataSource.setDriverClassName(driverClassName);
+        dataSource.setUrl(url);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        return dataSource;
+    }
+
+    @Bean
+    public LocalContainerEntityManagerFactoryBean entityManagerFactory() {
```
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/config/PersistenceConfig.java]: public LocalContainerEntityManagerFactoryBean entityManagerFactory(DataSource dataSource)

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/config/PostgreSQLConfig.java [hunk 3]
```diff
@@ -26,7 +26,7 @@ public class PostgreSQLConfig {
     @Value("${spring.datasource.url}")
     private String url;
 
-        private DataSource dataSource() {
+    private DataSource dataSource() {
```
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/config/PostgreSQLConfig.java]: This class is redundant. Please, delete this file.

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/config/H2Config.java [hunk 4]
```diff
@@ -1,34 +1,39 @@
...
 import javax.sql.DataSource;
 
 @Configuration
 @EnableTransactionManagement
-public class H2JdbcConfig {
-        @Profile("test")
+public class H2Config {
+    @Profile("test")
     public DataSource dataSource() {
         return new EmbeddedDatabaseBuilder()
```
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/config/H2Config.java]: Move this logic to PersistenceConfig class

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java [hunk 5]
```diff
@@ -1,78 +1,90 @@
...
 
 import static org.springframework.http.HttpStatus.CREATED;
-import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 @RestController
 @RequiredArgsConstructor
-@RequestMapping(value = "/certificates")
+@RequestMapping(value = "/api/certificates")
```
Comment (sslichnyi) [inline Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java]: '/api' - path can be applied the hole project with 'server.servlet.context-path=/api' configuration in application.properties file. Please take a look on it

**Review by sslichnyi — CHANGES_REQUESTED** [review]

File: Module#03/api/src/main/java/com/epam/esm/config/PersistenceConfig.java (hunk 2 above)
Comment (Fat-Frumos) [reply Module#03/api/src/main/java/com/epam/esm/config/PersistenceConfig.java]: Injected dependency dataSource:  
 - EntityManager Factory,  
 - DataSource Initializer;

File: Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java (hunk 5 above)
Comment (Fat-Frumos) [reply Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java]: config servlet.context path was added

**Comment by sslichnyi** [comment]
Please, create the following structure:
modules:
- model (entities, dto)
- repository-api (custom repository interfaces)
- repository-impl (bean which implements repository-api's interfaces)
- service-api (custom server interfaces)
- service-impl (beans which implements service-api interfaces, exceptions)
- web (config, SpringBootApplication main class, properties, controllers, dtos, excpetion handler, exceptions)

**Pull request merged**